docs/ref/templates/api.txt

```diff
@@ -922,38 +922,189 @@ option. It uses each loader until a loader finds a match.
+            depend on this argument should be updated to no longer use it.
+
+    .. method:: get_contents(self, origin)
+
+        Returns the contents for a template given a ``LoaderOrigin`` instance.
+
+        This is where a filesystem loader would read contents from the
+        filesystem, or a database loader would read from the database. If a
+        matching template doesn't exist, this should raise a
+        ``TemplateDoesNotExist`` error.
+
+    .. method:: def get_template(self, template_name, skip=None, template_dirs=None):
+
+        Returns a ``Template`` object for a given ``template_name`` by looping
+        through results from ``get_template_sources`` and calling
+        ``get_contents``. This returns the first matching template. If no
+        template is found, ``TemplateSyntaxError`` is raised.
+
+        This method takes an optional ``skip`` argument. ``skip`` is a list of
+        origins to ignore when extending templates. This allow templates to
+        extend other templates of the same name. It also used to avoid
+        recursion errors.
+
+        In general, it is enough to define ``get_template_sources`` and
+        ``get_contents`` for custom template loaders. ``get_template`` will
+        usually not need to be overridden.
+
+        .. warning::
+
+            The ``template_dirs`` argument has been deprecated. Loaders that
+            depend on this argument should be updated to no longer use it.
+
+    .. method:: def load_template_source(self, template_name, template_dirs=None)
+
+        Returns a tuple of (``template_string``, ``template_origin``), where
+        ``template_string`` is a string containing the template contents,
+        and ``template_origin`` is a string identifying the template source.
+        A filesystem-based loader may return the full path to the file as the
+        ``template_origin``, for example.
+
+        ``template_dirs`` is an optional argument used to control which
+        directories the loader will search in.
+
+        This method is called automatically by ``load_template`` and should
+        be overridden when writing custom template loaders.
+
+        .. deprecated:: 1.9
+
+         Custom loaders should use ``get_template`` and ``get_contents``
+         instead.
+
+    .. method:: def load_template(self, template_name, template_dirs=None)
+
+        Returns a tuple of (``template``, ``template_origin``), where ``template``
+        is a ``Template`` object and ``template_origin`` is a string identifying
+        the template source. A filesystem-based loader may return the full
+        path to the file as the ``template_origin``, for example.
+
+        .. deprecated:: 1.9
+
+         Custom loaders should use ``get_template`` and ``get_contents``
+         instead.
+
+Example
+-------
+
+A simple loader is one that loads templates from a Python dictionary. An
+implementation of such a loader would look like:
```

Comment:
That's basically the `LocMem` loader. It is worth repeating the code in the docs? What about suggesting to look at the implementation of built-in loaders instead?